common/presenters/table/object-to-table-head.js

```diff
@@ -1,17 +1,49 @@
 const { pickBy } = require('lodash')
+const queryString = require('query-string')
 
 const i18n = require('../../../config/i18n')
 
-function objectToTableHead({ head } = {}) {
-  if (!head) {
-    return undefined
+function buildUrl(head) {
+  const { query, key } = head
+  return queryString.stringify({
+    ...query,
+    sortBy: key,
+    sortDirection: (query && query.sortDirection) === 'asc' ? 'desc' : 'asc',
+  })
+}
+
+function buildWrapper(head) {
+  const label = i18n.t(head.html)
+  const url = buildUrl(head)
+  if (head.sortable) {
+    return `<a class="sortable-table__button" role="button" href="?${url}">${label}</a>`
   }
+  return label
+}
 
-  return pickBy({
-    ...head,
-    html: head.html ? i18n.t(head.html) : undefined,
-    text: head.text ? i18n.t(head.text) : undefined,
-  })
+function objectToTableHead(query) {
+  return function objectToTableHead({ head } = {}) {
+    if (!head) {
+      return undefined
+    }
+    if (head.sortable) {
+      head.query = query
+      head.attributes = {
+        ...head.attributes,
```

Comment:
What about just assigning this property in the case a column is sortable instead of re-assigning the attributes object:

```javascript
const { isSortable, sortKey } = head

if (isSortable) {
  const isCurrentSortColumn = sortKey === query.sortBy
  head.attributes['aria-sort'] = isCurrentSortColumn
    ? query.sortDirection === 'asc'
      ? 'ascending'
      : 'descending'
    : 'none',
}
```

I also find creating a variable (`isCurrentSortColumn`) for some booleans in a ternary can make the code a bit easier to read.

Reply:
The current version not only assigns the property, but also initialises `attributes` if it's `undefined`.

Reply:
The variable is a good point, and I have added it.